pytorch_lightning/accelerators/cpu.py

```diff
@@ -39,19 +39,16 @@ def get_device_stats(self, device: _DEVICE) -> Dict[str, Any]:
         return {}
 
     @staticmethod
-    def parse_devices(devices: Union[int, str, List[int]]) -> Union[int, str, List[int]]:
+    def parse_devices(devices: Union[int, str, List[int]]) -> int:
         """Accelerator device parsing logic."""
+        devices = device_parser.parse_cpu_cores(devices)
+        assert devices is not None
         return devices
 
     @staticmethod
-    def get_parallel_devices(devices: Union[int, str, List[int]]) -> List[torch.device]:
+    def get_parallel_devices(devices: int) -> List[torch.device]:
         """Gets parallel devices for the Accelerator."""
-        if isinstance(devices, int):
-            return [torch.device("cpu")] * devices
-        rank_zero_warn(
-            f"The flag `devices` must be an int with `accelerator='cpu'`, got `devices={devices!r}` instead."
-        )
-        return []
+        return [torch.device("cpu")] * devices
```

Comment:
this should independently validate that devices is a positive integer as well, right? what if someone doesn't call parse_devices before calling get_parallel_devices ?

Reply:
well, this can happen anywhere, even in `gpu.py`, `tpu.py`.. etc. I guess the assumption here was that parse_devices should be called first..
or we can parse them again inside `get_parallel_devices` for all?